crmd/utils.c

```diff
@@ -1008,6 +1008,55 @@ erase_status_tag(const char *uname, const char *tag, int options)
     }
 }
 
+static void
+crm_set_join_state(const char *uname, const char *start_state)
+{
+    if (safe_str_eq(start_state, "standby")) {
+        crm_notice("Starting node in %s state (%s)", start_state, uname);
+        update_attrd(uname, XML_CIB_ATTR_STANDBY, "on", NULL, FALSE);
+    } else if (safe_str_eq(start_state, "online")) {
+        crm_notice("Starting node in %s state (%s)", start_state, uname);
+        update_attrd(uname, XML_CIB_ATTR_STANDBY, "off", NULL, FALSE);
+    } else if (safe_str_eq(start_state, "default")) {
+        crm_notice("Starting node by default (%s)", uname);
+    } else {
+        crm_warn("Unrecognized start state '%s', using 'default' (%s)", start_state, uname);
+    }
+}
+
+static void
+init_attrs_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void *user_data)
+{
+    char **data = (char**)user_data;
+    char *uname = data[0];
+    char *start_state = data[1];
+    char *xpath = data[2];
+
+    do_crm_log_unlikely(rc == 0 ? LOG_DEBUG : LOG_NOTICE,
+                        "Erasing of \"%s\": %s (rc=%d)", xpath, pcmk_strerror(rc), rc);
+    crm_set_join_state(uname, start_state);
+    free(data[0]);
+    free(data[1]);
+    free(data[2]);
```

Comment:
`erase_xpath_callback(...)` doesn't have any free functions. We was sure that the memory allocated for `user_data` would be released further. We think `free_func` in `register_callback_full(...)` implements this.